pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -66,36 +69,78 @@ func (am *Alertmanager) Run(ctx context.Context) error {
 		return errors.Wrap(err, "unable to initialize the silencing component of alerting")
 	}
 
-	am.notificationLog, err = nflog.New(
-		nflog.WithRetention(time.Hour*24),                         //TODO: This is a setting.
-		nflog.WithSnapshot(filepath.Join("dir", "notifications")), //TODO: This should be a setting
-	)
-	if err != nil {
-		return errors.Wrap(err, "unable to initialize the notification log component of alerting")
+	return nil
+}
+
+func (am *Alertmanager) Run(ctx context.Context) error {
+	// Make sure dispatcher starts. We can tolerate future reload failures.
+	if err := am.ReloadConfigFromDatabase(); err != nil {
+		return err
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			am.stopDispatcher()
+			return nil
+		case <-time.After(1 * time.Minute):
```

Comment:
I'm not sure I want this right now, if if you are, I'd expect us not to no-op if the config nor templates have changed and we should do this work now.

Reply:
I will skip this for now and have it in place when we have the logic to skip config on no change.